jni/lib/src/lang/jinteger.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: overridden_fields
+
+import '../accessors.dart';
+import '../jni.dart';
+import '../jvalues.dart';
+import '../third_party/generated_bindings.dart';
+import '../types.dart';
+import 'jnumber.dart';
+
+class JIntegerType extends JObjType<JInteger> {
+  const JIntegerType();
+
+  @override
+  String get signature => r"Ljava/lang/Integer;";
+
+  @override
+  JInteger fromRef(JObjectPtr ref) => JInteger.fromRef(ref);
+
+  @override
+  JObjType get superType => const JNumberType();
+
+  @override
+  final superCount = 2;
+
+  @override
+  int get hashCode => (JIntegerType).hashCode;
+
+  @override
+  bool operator ==(Object other) {
+    return other.runtimeType == JIntegerType && other is JIntegerType;
```

Comment:
Doesn't `other.runtimeType == JIntegerType` imply `other is JIntegerType`?

Reply:
other is Something could mean that other extends Something as well.

Reply:
Then runtimeType will be most concrete subtype right?

Reply:
Yes. But you can override runtimeType to be anything. So in reality there is no sure fire way to check if something is of an exact type but this is the closest we can get.

Reply:
I see that's very pythonic.